src/main/webapp/javascript/recipe-rendering.js

```diff
@@ -50,11 +51,77 @@ createRecipeCard = (divID, recipeInfo) => {
 }
 
 /**
- * Takes the id for a recipe in Datastore, and using the recipe page template
- * completes the page for the user. This function is added to each recipe card
- * name, and should be called onClick().
+ * Takes the id for a recipe in Datastore, and redirects to the recipe page template
+ * with the recipe ID as a search param.
+ */
+function redirectRecipePage(id) {
+  window.location.assign("/pages/RecipePageTemplate.jsp?id=" + id.toString());
+}
+
+/**
+ * Function for loading the recipe template page based on the id in the URL
+ * search params.
+ */
+async function fillRecipeTemplate() { 
+  const recipeId = (new URL(document.location)).searchParams.get("id");
+
+  // Gets the servlet data for the recipe
+  // TODO: put this code in an if statement to handle recipe not found scenarios
+  let response = await fetch("/retrieve-id?recipeId=" + recipeId.toString());
+  let responseText = await response.text();
+  console.log(responseText);
```

Comment:
Do you want to leave this in, or was it just for debugging? If you're keeping it, consider adding some context, like "Response from Id Search Servlet:" so it's obvious what it is without looking at the code.

Reply:
Ack and fixed, I removed it